torch_xla_py/data_parallel.py

```diff
@@ -84,9 +84,34 @@ def close(self):
       dqueue.queue.close()
       dqueue.loader_queue.close()
 
-  def _expand_sample_batch(self, data, target):
-    # TODO: Expand last sample,target to batch size
-    return data, target
+  def _get_batch_size(self, data, dim):
+    size = []
+
+    def fn(v):
+      csize = v.size()[dim]
+      if not size:
+        size.append(csize)
+      else:
+        assert csize == size[0]
+
+    xu.for_each_instance(data, torch.Tensor, fn)
+    return size[0] if size else None
+
+  def _expand_sample_batch(self, data):
+    # TODO: Expand last sample to batch size
```

Comment:
Why would we need this? I vaguely remember we've had a discussion before, but I don't remember the details.

Reply:
If the last batch size is not full, fill it up.
With XTen this would not be much of a problem, but if users expect batch-sized input data, the last partially-full eventual batch will have to be either discarded, or filled.

Reply:
I don't think an user would ever expect batch-sized input data, all code should be batch-size agnostic, otherwise it wouldn't be possible to use different batch sizes. Unless something would really get broken (not the user expectation), I think it's fine to remove the comment.

Reply:
DataLoader has a drop_last argument ... using same semantics.